.github/workflows/javascript-checks.yml

```diff
@@ -0,0 +1,54 @@
+name: "Javascript Checks"
+
+on:
+  pull_request:
+    branches:
+      - main
+
+    paths:
+      - ".github/workflows/javascript-checks.yml"
+      - "javascript/**"
+
+jobs:
+  Conformity-Check:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: ["10"]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: "Checking Code Conformity with Style Guidelines"
+        run: |
+          cd javascript
+          yarn
+          yarn lint:check
+        env:
+          CI: true
+
+  Unit-Tests-Check:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        node_version: ["10"]
```

Comment:
I believe we should at least use v12, and probably it will be nice to add an .nvmrc

Reply:
sure thing, for now for the sake of simplicity, will start with node 10, later will add more support to 12 and more, and will enforce that in configuration, thanks for noting that!